proposals/892-experiment-tracking/README.md

```diff
@@ -0,0 +1,514 @@
+   integration with Katib after the initial implementation, we are not targeting the decoupling of Katib's existing
+   experiment tracking system as part of this proposal. The initial focus is on the SDK and Pipelines experience.
+1. **MLFlow UI support**: Although the MLFlow plugin should enable MLFlow UI compatibility, providing direct MLFlow UI
+   support is not a targeted goal of this proposal.
+1. **Automatic experiment tracking from the Kubeflow Trainer**: While users can leverage the new functionality through
+   the Kubeflow SDK or MLFlow plugin, we are not implementing automatic experiment tracking directly from the Kubeflow
+   Trainer as part of this proposal. The community should revisit integrations at a later date.
+
+## Proposal
+
+### Model Registry Expansion
+
+Model Registry was designed to be a centralized metadata store for Kubeflow. It currently focuses on managing metadata
+for the lifecycle of machine learning models, from registration and versioning through deployment and serving. This
+proposal expands its scope by adding new metadata resource types to cover experiment tracking. As part of this
+expansion, we could consider renaming Model Registry to better reflect its broader scope.
+
+The expanded Model Registry will support:
+
+- **Experiments and Runs**: Centralized tracking of experiments and their associated runs across all Kubeflow components
+- **Multi-tenancy**: Namespace-level isolation using Kubernetes RBAC
+- **Artifact Management**: Metadata storage of models, datasets, and other ML artifacts
+- **Unified Metadata**: Consistent metadata schema across the entire Kubeflow ecosystem
+
+### Kubeflow Pipelines Integration
+
+The integration of Kubeflow Pipelines with the centralized experiment tracking system will be implemented in two phases:
+
+#### Phase 1: Dual Registration and Unified Experiment View
+
+- Enable Kubeflow Pipelines to register experiment metadata in both MLMD and Model Registry simultaneously
+- Provide users with a unified view of experiments across the entire Kubeflow ecosystem
+- Maintain backward compatibility with existing MLMD-dependent tooling and existing Kubeflow Pipelines UI
+
+#### Phase 2: Complete MLMD Decoupling
+
+- Eliminate the MLMD dependency by migrating all Kubeflow Pipelines (KFP) metadata storage to the KFP database and
+  external metadata trackers
+- Provide configuration options allowing users to choose between Model Registry and MLMD for experiment tracking
+- Kubeflow Pipelines UI enhancements with more direct integration with Model Registry if that mode is chosen
+
+### SDK Integration
+
+#### MLFlow SDK Compatibility
+
+- Implement MLFlow plugins to connect to the Model Registry backend
+- Enable users to leverage MLFlow's familiar APIs while storing data in centralized Kubeflow infrastructure
+- Support automatic logging for popular ML frameworks (TensorFlow, PyTorch, scikit-learn, etc.) through MLFlow's SDK
+
+#### Kubeflow SDK Enhancement
```

Comment:
It would be nice to have some code snippet on how those integration will look like.

Reply:
I'll add this in `Kubeflow SDK Implementation` under `Design Details`. The example is from a proof of concept developed by @kramaranya!